frontend/src/pages/main/payment/components/qrscan/QrScan.tsx

```diff
@@ -0,0 +1,72 @@
+import React, {useEffect} from 'react';
+import {useAppDispatch} from "../../../../../stores/redux/hooks";
+import {IQRCodeData} from "../../../../../types/QRCodeCredentials";
+import {Html5Qrcode} from "html5-qrcode";
+import {cameraConfig, cameraIdOrConfig, elementId} from "./utils";
+import {QrScanProps} from "./props";
+import _ from "lodash";
+import {showNotification} from "../../../../../stores/redux/notification/notificationSlice";
+import {NotificationType} from "../../../../../types/notification";
+import {Notifications} from "../../../../../common/constansts/notifications";
+import {Box} from "@mui/material";
+import "./styles.css";
+
+const QrScan: React.FC<QrScanProps> = ({updateQrCodeData}) => {
+
+    const dispatch = useAppDispatch();
+
+    const validateQrCodeData = (qrCodeData: IQRCodeData): boolean => {
+        try {
+            return !(_.isEmpty(qrCodeData) || _.isEmpty(qrCodeData.name) || qrCodeData.money <= 0 || qrCodeData.walletId < 0);
```

Comment:
I divided it on purpose because this structure looks complicated. But if it's ok for you, you can leave as it is